phpdotnet/phd/Package/Generic/XHTML.php

```diff
@@ -2039,6 +2045,110 @@ public function format_itemizedlist($open, $name, $attrs, $props) {
+            $list = "";
+            foreach ($this->cchunk["simplelist"]["members"] as $member) {
+                $list .= $member . ", ";
+            }
+            $list = rtrim($list, ", ");
+
+            $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+            return $list . '</span>';
+
+        }
+
+        if ($this->cchunk["simplelist"]["type"] === "horiz") {
+
+            $table = "";
+            for ($i = 0; $i < count($this->cchunk["simplelist"]["members"]); $i++) {
+                if ($i % $this->cchunk["simplelist"]["columns"] === 0) {
+                    $table .= str_repeat(" ", $props["depth"] + 2) . "<tr>\n";
+                }
+
+                $table .= str_repeat(" ", $props["depth"] + 3) . "<td>" . $this->cchunk["simplelist"]["members"][$i] . "</td>\n";
+
+                if ($i % $this->cchunk["simplelist"]["columns"] === $this->cchunk["simplelist"]["columns"] - 1) {
+                    $table .= str_repeat(" ", $props["depth"] + 2) . "</tr>\n";
+                }
+            }
+            if ($i % $this->cchunk["simplelist"]["columns"] !== 0) {
+                $numOfMissingCells = $this->cchunk["simplelist"]["columns"] - ($i % $this->cchunk["simplelist"]["columns"]);
+                $oneRow = str_repeat(" ", $props["depth"] + 3) . "<td></td>\n";
+                $table .= str_repeat($oneRow, $numOfMissingCells);
+                $table .= str_repeat(" ", $props["depth"] + 2) . "</tr>\n";
+            }
+
+            $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+
+            return $table . str_repeat(" ", $props["depth"] + 1) . "</tbody>\n" . str_repeat(" ", $props["depth"]) . "</table>";
+
+        }
+
+        if ($this->cchunk["simplelist"]["type"] === "vert") {
+
+            $table = "";
+            $numOfRows = ceil(count($this->cchunk["simplelist"]["members"]) / $this->cchunk["simplelist"]["columns"]);
+            for ($row = 0; $row < $numOfRows; $row++) {
+                $table .= str_repeat(" ", $props["depth"] + 2) . "<tr>\n";
+                for ($col = 0; $col < $this->cchunk["simplelist"]["columns"]; $col++) {
+                    $memberIndex = ($numOfRows * $col) + $row;
+                    $table .= str_repeat(" ", $props["depth"] + 3) . "<td>";
+                    if ($memberIndex < count($this->cchunk["simplelist"]["members"])) {
+                        $table .= $this->cchunk["simplelist"]["members"][$memberIndex];
+                    }
+                    $table .= "</td>\n";
+                }
+                $table .= str_repeat(" ", $props["depth"] + 2) . "</tr>\n";
+            }
+            $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+
+            return $table . str_repeat(" ", $props["depth"] + 1) . "</tbody>\n" . str_repeat(" ", $props["depth"]) . "</table>";
+        }
+
+        $this->cchunk["simplelist"] = $this->dchunk["simplelist"];
+        return '</ul>';
```

Comment:
How about separating each case into private methods, and doing a `return match($this->cchunk["simplelist"]["type"]) {};` on the attribute value to return the value from the private methods?

Reply:
Sounds good.